### DISCUSSION TITLE
dace ignoring auto_optimize flag when using to_sdfg?

### DISCUSSION THREAD

**Discussion by rohanrayan** [spcl/dace#1379] [Q&A] [answered]
I am not sure if this is a bug or a misunderstanding from my part, I have tested this on v0.14.4 of dace. 
I have a simple function below:

**@dace.program(auto_optimize=True)
def madd(A, B, m, n):
    for i in range(0, m):
        for j in range(0, n):
            A[i][j] += B[i][j]
      return A**

I define example numpy matrices as follows:

**m = 200
n = 250
vA = np.random.rand(m, n)
vB = np.random.rand(m, n)**

When I try to generate the SDFG using the following commands:

**madd_dace_sdfg = madd.to_sdfg(vA, vB, m, n)
madd_dace_compiled = madd_dace_sdfg.compile()**

I see that the auto_optimize flag for the function gets ignored i.e. it produces the same graph regardless of whether the auto_optimize flag is set to True or False.

On the other hand, if I generate the graph using the following command (without going through the intermediate to_sdfg(): 

**madd_dace_compiled = madd.compile(vA, vB, m, n)**

I see that the graph changes based on the auto_optimize flag. 

Is there something I am missing?

Thanks in advance!

**Comment by tbennun** [comment, marked as answer]
Hi, thank you for noting this! This behavior is by design, but `to_sdfg` should raise a warning and it does not. Could you please report an issue? This should be a two-line fix in https://github.com/spcl/dace/blob/master/dace/frontend/python/parser.py#L235

**Reply by rohanrayan** [reply]
Is there any documentation where these flags and their uses are documented? I didnt find this in the standard documentation page.

**Reply by tbennun** [reply]
Of course:
* `@dace.program`: https://spcldace.readthedocs.io/en/latest/source/dace.frontend.python.html#dace.frontend.python.interface.program
* Auto-optimize: https://spcldace.readthedocs.io/en/latest/sdfg/auto_optimize.html#opt-auto
* Recompile/regenerate code: https://spcldace.readthedocs.io/en/latest/general/debugging.html#debugging-and-recompiling-generated-code

In order to improve our documentation, which ones were you looking for and where/how did you search for them?

**Reply by rohanrayan** [reply]
Thanks for the links. I was interested more in the design decisions which were taken. Because as a beginner to this framework, it is not very obvious to me why the to_sdfg() should ignore the auto optimize flag for example. But maybe it is explained somewhere in the documentation and I just missed it!
